apps/emails-and-messages/src/modules/smtp/configuration/smtp-metadata-manager.ts

```diff
@@ -1,12 +1,12 @@
 import { SettingsManager } from "@saleor/app-sdk/settings-manager";
 import { SmtpConfig } from "./smtp-config-schema";
 
-export interface SmtpConfigurator {
-  setConfig(config: SmtpConfig): Promise<void>;
-  getConfig(): Promise<SmtpConfig | undefined>;
+export interface GenericConfigurator<T> {
+  setConfig(config: T): Promise<void>;
+  getConfig(): Promise<T | undefined>;
 }
```

Comment:
do you even need this interface? You don't inject managers dynamically, you need them to be statically available everywhere, I don't see a reason of having them and fighting with generics

Reply:
Removed